syntax = "proto3";

package buaa.file;

//服务端向客户端发送规则库列表项
message rule_file_info{
    bytes file_name = 1;//文件名
    bytes file_desc = 2;//文件描述
    bytes create_time = 3;//文件创建时间
}


//客户端回调欲加载文件列表
message rule_file_chk{
    bool ischeck = 1; //是否加载规则库
    bytes filename = 2; //加载的文件名称
}


//解析加载的规则库规则信息
message rule_names{
    bytes rule_name = 1;
}

//解析加载的规则库规则信息
message fact_names{
    bytes fact_name = 1;
}

//服务端向客户端发送知识库文件列表项
message fact_file_info{
    bytes file_name = 1;//文件名
    bytes file_desc = 2;//文件描述
    bytes create_time = 3;//文件创建时间
}

//客户端回调欲加载知识文件列表
message fact_file_chk{
    bool ischeck = 1; //是否加载规则库
    bytes filename = 2; //加载的文件名称
}


//事实库模板列表
message fact_list{
    bytes template_name = 1; //模板名称
    bytes slot_name = 2;    //槽名称
}

//可选变量
message fact_var{
    bytes file = 1; //所属事实库文件名称
    bytes var_name = 2; //变量名称
    bytes var_val = 3; //变量值
}

//slot variable 映射关系
message mapper{
    bytes template = 1;//所属模板名称
    bytes slot = 2; //槽名称
    bytes variable = 3; //变量名称
}